server/src-lib/Hasura/RQL/DML/Mutation.hs

```diff
@@ -111,9 +111,11 @@ mkSelCTEFromColVals parseFn qt allCols colVals =
         }
   where
     tableAls = S.Alias $ Iden $ snakeCaseQualObject qt
-    colNames = map pgiColumn allCols
+    sortedCols = flip sortBy allCols $ \lCol rCol ->
+                 compare (pgiOrdinalPosition lCol) (pgiOrdinalPosition rCol)
+    colNames = map pgiColumn sortedCols
     mkTupsFromColVal colVal =
-      fmap S.TupleExp $ forM allCols $ \ci -> do
+      fmap S.TupleExp $ forM sortedCols $ \ci -> do
```

Comment:
Could you please add a comment here that explains why it’s important that the columns be sorted here? Ideally with an example SQL query that illustrates what goes wrong if they are not sorted.